lib/handlers/router-entities/pool-caching/v3/cache-dynamo-pool.ts

```diff
@@ -0,0 +1,78 @@
+import { DynamoCaching, DynamoCachingProps } from '../../cache-dynamo'
+import { Pool } from '@uniswap/v3-sdk'
+import { log } from '@uniswap/smart-order-router'
+
+interface DynamoCachingV3PoolProps extends DynamoCachingProps {}
+
+export class DynamoCachingV3Pool extends DynamoCaching<string, number, Pool> {
+  constructor({ tableName }: DynamoCachingV3PoolProps) {
+    super({ tableName })
+  }
+
+  override async get(partitionKey: string, sortKey?: number): Promise<Pool | undefined> {
+    if (sortKey) {
+      const getParams = {
+        TableName: this.tableName,
+        Key: {
+          poolAddress: partitionKey,
+          blockNumber: sortKey,
```

Comment:
aw what's the use case for fetching some past blocks? Is it when we don't know what's the latest the block number?

Reply:
In case we want to optimistically use the previous block data, but I think we can go with what we have right now, and then see how it performs before making other changes :)